### PR TITLE
Add implementation to TextProcessorJDK.topVowelAndConsonant() for haiku-kata-solutions module #296

### LINKED ISSUE #296: Add implementation to TextProcessorJDK.topVowelAndConsonant() for haiku-kata-solutions module
The method `topVowelAndConsonant()` in the `TextProcessorJDK` class for the [haiku-kata-solutions](https://github.com/eclipse/eclipse-collections-kata/tree/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions) module is not yet implemented. Follow the TODOs and hints given in the [method](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java#L95). Please do not add solutions to the non-solution modules. After writing the code, uncomment the test and tag annotations and check that the [test case](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions/src/test/java/org/eclipse/collections/haikukata/TextProcessorJDKTest.java#L56) `topVowelAndConsonant()` in the `TextProcessorJDKTest` passes for the [haiku-kata-solutions](https://github.com/eclipse/eclipse-collections-kata/tree/52070f7651667c27e974e45eea2cce4309345530/haiku-kata-solutions) module. Make sure to adhere to our [contribution guide](https://github.com/eclipse/eclipse-collections-kata/blob/52070f7651667c27e974e45eea2cce4309345530/CONTRIBUTING.md). 

Please note that this issue is created for GHC 2023 Open Source Day, thus priority will be given to the attendees of this event. After the event, this issue will be open to anyone.

### PR DESCRIPTION
**Pull request by rsp2210**
Fix #296

### CODE REVIEW

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 1]
```diff
@@ -94,36 +94,43 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
-
-        // TODO: Find the top consonant
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top consonant Character value using getKey on Map.Entry.
-        char topConsonant = 'b';
+    public TopVowelAndConsonant topVowelAndConsonant() {
+        // Find all of the alphabetic letters from the haiku, convert them to lowercase,
+        // and count the Character values in a Map.
+        Map<Character, Long> charCounts = this.getHaikuAsChars()
+                .stream()
```
Comment (rzrobin213) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:101]: IntStream does not have the method .stream() so this code doesn't compile. You already have an IntStream, you just need to convert it into a stream of character objects. I would take a look at the IntStream.mapToObj([IntFunction](https://docs.oracle.com/javase/8/docs/api/java/util/function/IntFunction.html)<? extends U> mapper) function.

https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 2]
```diff
@@ -94,36 +94,43 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
+
+        // Create a List of the top 26 occurrences in Map.Entry<Character, Long> instances sorted in reverse order by count.
+        List<Map.Entry<Character, Long>> entries = charCounts.entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .toList();
+
+        // Find the top vowel.
+        char topVowel = entries.stream()
+                .filter(this::isVowel)
```
Comment (rzrobin213) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:114]: Sorry for being unclear. I think you need to .getKey() to use the isVowel method. `entries` is a list of map objects and isVowel only has aa char as input. So, using a lambda with .getKey on the map makes sense to me, just like how you had in the previous one.

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java [hunk 3]
```diff
@@ -94,36 +94,43 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
...
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("No vowels found in haiku"))
+                .getKey();
+
+        // Find the top consonant.
+        char topConsonant = entries.stream()
+                .filter(!this::isvowel)
```
Comment (rzrobin213) [inline haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java:121]: Same comment as above; I believe you had it right in the last time. I just wanted to get you to change the previous one. Sorry for the confusion.

**Review by rzrobin213 — CHANGES_REQUESTED** [review]
Few changes!

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java (hunk 1 above)
Comment (rsp2210) [reply haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java]: I worked on the solution as per your suggestions. Thanks for the link, It was helpful in understanding the function better..

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java (hunk 2 above)
Comment (rsp2210) [reply haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java]: No worries. I made the necessary adjustments and updated the code accordingly. Thanks again for catching that.

File: haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java (hunk 3 above)
Comment (rsp2210) [reply haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java]: No need to apologize; I appreciate your input and feedback!

**Review by rzrobin213 — APPROVED** [review]
Looks good to me! Nice work!

**Comment by rsp2210** [comment]
> Looks good to me! Nice work!

**Comment by rsp2210** [comment]
Thank you! I closed it by mistake, I reopened it, sorry.

**Comment by prathasirisha** [comment]
Thank you for your contribution @rsp2210 and thank you, @rzrobin213 for the code review! Good work.

**Pull request merged**